core/src/main/scala/io/projectglow/bgen/BgenSchemaInferrer.scala

```diff
@@ -28,24 +28,33 @@ import io.projectglow.sql.util.SerializableConfiguration
  * Infers the schema of a set of BGEN files from the user-provided options and the header of each
  * file.
  *
- * Currently the implementation is very simple. It checks if sample IDs are defined in either a
- * .sample file or in any of the headers. If so, it returns a fixed schema that includes a
- * `sampleId` genotype field. If not, it returns the same schema without a `sampleId` field.
+ * - If sample IDs are defined in either a .sample file or in any of the headers,
+ *   `sampleId` is added to the genotype field.
+ * - If a hard call threshold is specified, `calls` is added to the genotype field.
  */
 object BgenSchemaInferrer {
   def inferSchema(
       spark: SparkSession,
       files: Seq[FileStatus],
       options: Map[String, String]): StructType = {
+    val hasSampleIds = includeSampleIds(spark, files, options)
+    val hasHardCalls = options.contains(BgenOptions.HARD_CALL_THRESHOLD)
```

Comment:
Hm, I feel like we should include hard calls by default. AFAIK, every user so far has wanted hard calls when reading BGEN files. We could add a separate flag to disable, or accept -1 here.